### PR TITLE
begin a test framework so we can collect exhaustive test vectors

### PR DESCRIPTION
**Pull request by lavalamp**
It seems like we don't have something like this. I haven't done any hard work here yet--perhaps if we commit this we can do that in parallel?

### CODE REVIEW

File: staging/src/k8s.io/smd/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Structured Merge and Diff
```
Comment (apelisse) [inline staging/src/k8s.io/smd/README.md]: Can we change the name of the package? `mergediff`? Something that is more easily to guess if you've never seen the full name first.

**Review by kwiesmueller — CHANGES_REQUESTED** [review]

File: staging/src/k8s.io/smd/testvectors/listordering.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+- b
+- c
+- d`),
+			LastObject: YAMLObject(`
+items:
+- a
+- b
+- c
+`),
+			LiveObject: YAMLObject(`
```
[the remark this answers is not in the document]
Comment (jennybuckley) [reply staging/src/k8s.io/smd/testvectors/listordering.go]: Should probably be "NewObject" here

**Comment by lavalamp** [comment]
Update pushed, addressed comments and made it a lot better in other ways. PTAL

**Comment by lavalamp** [comment]
I'm trying to figure out if this really must go in staging or if I can put it in its own repo. I think it depends on what type I use to represent a value and an OpenAPI schema--will those types come from k/k or staging dir?

Schema I suspect not. Value might plausibly be unstructured. Need to look into that.

**Comment by lavalamp** [comment]
Looks like today's unstructured offers the ability to extract its contents in the form of map[string]interface{}, so I can be interoperable without actually needing to take on a dependency.

I'll (sigh) look into getting a separate repo for this.

**Comment by lavalamp** [comment]
Moved to https://github.com/kubernetes-sigs/structured-merge-diff/pull/2

**Pull request closed without merging**